src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,158 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(head, value);
+        if (tail == null) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            head.next = newNode;
+            head = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Objects.checkIndex(index,size + 1);
```

Comment:
please don't use class Objects, see the common mistakes:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#dont-use-class-objects